src/devices/src/virtio/net/iovec.rs

```diff
@@ -143,7 +143,7 @@ impl IoVecBuffer {
             //    to guest physical memory, whereas `buf_ptr` to Firecracaker-owned memory.
             unsafe {
                 let seg_ptr = (seg.iov_base as *const u8).add(start);
-                std::ptr::copy_nonoverlapping(seg_ptr, buf_ptr, buf_end - buf_start + 1);
+                std::ptr::copy_nonoverlapping(seg_ptr, buf_ptr, buf_end - buf_start);
             }
```

Comment:
I think it would be safer to do something like:
```rust
let slice = &mut buf[buf_start..buf_end];
slice .copy_from_slice(std::slice::from_raw_parts(seg_ptr, slice.len()));
```